assets/differentialabundance_report.Rmd

```diff
@@ -815,17 +815,23 @@ if (any(unlist(params[paste0(possible_gene_set_methods, '_run')]))){
     if (unlist(params[paste0(gene_set_method, '_run')])){
       cat("\n### ", toupper(gene_set_method) ," {.tabset}\n")
       
-      reference_gsea_tables <- paste0(contrasts$id, '.gsea_report_for_', contrasts$reference, '.tsv')
-      target_gsea_tables <- paste0(contrasts$id, '.gsea_report_for_', contrasts$target, '.tsv')
-
-      for (i in 1:nrow(contrasts)){
-        cat("\n#### ", contrast_descriptions[i], "\n")
+      for (gmt_file in strsplit(params$gsea_gene_sets, " ")[[1]]) {
+        gmt_name <- strsplit(basename(gmt_file), '.', fixed=T)[[1]]
+        gmt_name <- paste0(head(gmt_name, -1), collapse=".")
```

Comment:
```suggestion
        gmt_name <- basename(gmt_file)
```

Sorry, I'm not sure what you were trying to do there, but it seems like a plain `basename` does the same thing

Reply:
Hmm, that was to remove the file extension. I suppose we can leave it in though

Reply:
```
> basename(tools::file_path_sans_ext('s3://fee/fi/fo.txt.fi'))
[1] "fo.txt"
```
?